python/pyproject.toml

```diff
@@ -18,10 +18,8 @@
 [build-system]
 requires = [
     "cython >= 0.29",
-    "numpy==1.16.6; python_version<'3.9'",
-    "numpy==1.19.4; python_version=='3.9'",
-    "numpy==1.21.3; python_version>'3.9'",
-    "setuptools",
+    "oldest-supported-numpy",
```

Comment:
We may need to add `numpy>=1.16.6` here as well.

Reply:
Yes, there are older numpy versions pinned for python < 3.8: https://github.com/scipy/oldest-supported-numpy/blob/master/setup.cfg#L44-L46

Reply:
We don't support numpy <= 1.16.6? Do you remember the reason?

Reply:
Based on the blame it points to https://github.com/numpy/numpy/issues/17913.

I guess the latest crossbow builds should fail if the pin is still required.

Reply:
`oldest-supported-numpy` should be >=1.17, so no need to enforce this?

Reply:
I think we already worked around the issue: https://github.com/apache/arrow/pull/8834

Reply:
Yeah, I searched for `PyArray_DescrCheck` across the codebase and didn't find any occurrences, so hope that pinning is not necessary.

Reply:
I wonder what do we need in `install_requires`: https://github.com/apache/arrow/blob/master/python/setup.py#L572

Reply:
Let's not change it if it doesn't annoy anyone.

Reply:
> I wonder what do we need in install_requires: https://github.com/apache/arrow/blob/master/python/setup.py#L572

Yes, I don't think this needs to be changed. `oldest-supported-numpy` is about build dependencies, not runtime install dependencies